algo/src/main/java/org/neo4j/graphalgo/UnionFindProc.java

```diff
@@ -0,0 +1,64 @@
+package org.neo4j.graphalgo;
+
+import algo.Pools;
+import org.neo4j.graphalgo.api.Graph;
+import org.neo4j.graphalgo.core.GraphLoader;
+import org.neo4j.graphalgo.core.heavyweight.HeavyGraphFactory;
+import org.neo4j.graphalgo.core.utils.dss.DisjointSetStruct;
+import org.neo4j.graphalgo.impl.GraphUnionFind;
+import org.neo4j.kernel.internal.GraphDatabaseAPI;
+import org.neo4j.logging.Log;
+import org.neo4j.procedure.*;
+
+import java.util.stream.Stream;
+
+/**
+ * @author mknblch
+ */
+public class UnionFindProc {
+
+    @Context
+    public GraphDatabaseAPI api;
+
+    @Context
+    public Log log;
+
+    @Procedure(value = "org.neo4j.graphalgo.unionFind")
+    @Description("CALL org.neo4j.graphalgo.unionFind(label:String, relationship:String) " +
+            "YIELD nodeId, setId - yields a setId to each node id")
+    public Stream<DisjointSetStruct.Result> unionFind(
```

Comment:
same for all other algorithms:

default values for label and rel-type of null (empty string), to run the algo across the whole graph

can be done later after this is merged